### PR TITLE
Add image-routes, controllers and helpers

### PR DESCRIPTION
**Pull request by carpecodeum**
# Description

Add images routes controllers and helpers.

Fixes #424 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

Annotation 2020-06-18 162854

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: labellab-flask/api/main.py [hunk 1]
```diff
@@ -5,10 +5,9 @@
 from flask_jwt_extended import JWTManager
 
 from api.config import config
-from api.routes import users, models
-from api.extensions import db, migrate, jwt
-from api.models import User, Image, Label, LabelData, ProjectMembers, Projects, Team, RevokedToken, MLModel
-
+from api.routes import users, projects, models, images
```
Comment (UdeshUK) [inline labellab-flask/api/main.py:8]: ```project``` module is not found. Project routes are not yet defined, right?

File: labellab-flask/api/controllers/imagescontroller.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        Handle POST request for this view.
+        Url --> /api/v1/image/create/<int:project_id>
+        """
+        # getting JSON data from request
+        post_data = request.get_json(silent=True,
+                                     force=True)
+        current_user = get_jwt_identity()
+        try:
+            images = post_data["images"]
+            image_names = post_data["imageNames"]
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/imagescontroller.py:36]: Change ```imagesNames``` to snake case to match the other endpoints.

File: labellab-flask/api/controllers/imagescontroller.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            }
+            return make_response(jsonify(response)), 201
+
+        except Exception as err:
+            print("Error occured: ", err)
+            response = {
+                "success": False,
+                "msg": "Could not save images to the server."
+            }
+            return make_response(jsonify(response)), 401
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/imagescontroller.py:84]: Better to send a 500 status here

File: labellab-flask/api/main.py (hunk 1 above)
Comment (carpecodeum) [reply labellab-flask/api/main.py:8]: Project routes are defined in another pull request..:). If you want I can remove it from here

File: labellab-flask/api/controllers/imagescontroller.py (hunk 2 above)
Comment (carpecodeum) [reply labellab-flask/api/controllers/imagescontroller.py:36]: sure !

File: labellab-flask/api/controllers/imagescontroller.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+        except KeyError as err:
+            response = {
+                "success": False,
+                "msg": f'{str(err)} key is not present'
+            }
+            return make_response(jsonify(response)), 400
+
+        project_id = project_id
+        user_id = current_user
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/imagescontroller.py:47]: its current_user.id right?

**Comment by ivantha** [comment]
@carpecodeum There are some conflicts. Can you please rebase so that we can merge this?

File: labellab-flask/api/controllers/imagescontroller.py (hunk 4 above)
Comment (carpecodeum) [reply labellab-flask/api/controllers/imagescontroller.py:47]: no no current_user is the user_id only 🙂

**Comment by carpecodeum** [comment]
> @carpecodeum There are some conflicts. Can you please rebase so that we can merge this?

done!

File: labellab-flask/api/controllers/imagescontroller.py (hunk 4 above)
Comment (hiranthaR) [reply labellab-flask/api/controllers/imagescontroller.py:47]: oh ok

**Pull request merged**